lib/config/index.js

```diff
@@ -136,4 +138,30 @@ function checkLimitsConfig (cfg) {
   }
 }
 
+/** Check the format of the error-codes config files
+ *
+ */
+
+function checkErrorCodes() {
+
+  for (const domain in errorCodes) {
+    const name = errorCodes[domain];
+    if (!name.hasOwnProperty('subdomains') || !name.hasOwnProperty('code')) {
+      throw new KuzzleInternalError(`Invalid configuration: Missing mandatory field in ${domain} error-codes config file.`, name, 0);
+    }
+    for (const subdomain in errorCodes[domain]['subdomains']) {
+      const name = errorCodes[domain]['subdomains'][subdomain];
+      if (!name.hasOwnProperty('errors') || !name.hasOwnProperty('code')) {
+        throw new KuzzleInternalError(`Invalid configuration: Missing mandatory field in ${domain} error-codes config file.`, name, 0);
+      }
+      for (const error in errorCodes[domain]['subdomains'][subdomain]['errors']) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Restrict what this loop acts on by testing each property. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AForIn)